hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/replication/TestVerifyReplication.java

```diff
@@ -447,6 +447,32 @@ public void testVerifyRepJobWithPeerTableNameAndSnapshotSupport() throws Excepti
     checkRestoreTmpDir(CONF2, tmpPath2, 2);
   }
 
+  @Test
+  public void testVerifyReplicationThreadedRecompares() throws Exception {
+    // Populate the tables with same data
+    runBatchCopyTest();
+
+    // ONLY_IN_PEER_TABLE_ROWS
+    Put put = new Put(Bytes.toBytes(Integer.toString(NB_ROWS_IN_BATCH)));
+    put.addColumn(noRepfamName, row, row);
+    htable3.put(put);
+
+    // CONTENT_DIFFERENT_ROWS
+    put = new Put(Bytes.toBytes(Integer.toString(NB_ROWS_IN_BATCH - 1)));
+    put.addColumn(noRepfamName, row, Bytes.toBytes("diff value"));
+    htable3.put(put);
+
+    // ONLY_IN_SOURCE_TABLE_ROWS
+    put = new Put(Bytes.toBytes(Integer.toString(NB_ROWS_IN_BATCH + 1)));
+    put.addColumn(noRepfamName, row, row);
+    htable1.put(put);
+
+    String[] args = new String[] { "--reCompareThreads=10", "--reCompareTries=3",
+      "--recomparesleep=1", "--peerTableName=" + peerTableName.getNameAsString(),
+      UTIL2.getClusterKey(), tableName.getNameAsString() };
+    runVerifyReplication(args, NB_ROWS_IN_BATCH - 1, 3);
+  }
+
   @AfterClass
   public static void tearDownAfterClass() throws Exception {
```

Comment:
im sort of concerned about the problem i mentioned in `cleanup` where we could end the job with outstanding recompares. Can we add a test for that? One thought i had would be to set things up identically to above, but pass in a very long recompareSleep and only 1 recompareThreads. So i think that will guarantee to have the job finish with at least 1 runnable still running, and possibly some runnables still queued (depending on how many mappers there are). We can then validate that those count as BADROWS and FAILED_RECOMPAREs